src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TeleporterSystem.java

```diff
@@ -123,8 +128,28 @@ private String setPlayerTeamToTeleporterTeam(EntityRef player, EntityRef telepor
 
     private void handlePlayerTeleport(EntityRef player, String team) {
         Vector3f randomVector = new Vector3f(-1 + random.nextInt(3), 0, -1 + random.nextInt(3));
+        player.send(new PregameEvent());
         player.send(new CharacterTeleportEvent(randomVector.add(LASUtils.getTeleportDestination(team))));
         player.addOrSaveComponent(startingInventory);
         player.send(new RequestInventoryEvent(startingInventory.items));
+        sendGameStartMessageEventToClients();
+    }
+
+    private void sendTimerEventToClients() {
+        if (entityManager.getCountOfEntitiesWith(ClientComponent.class) != 0) {
+            Iterable<EntityRef> clients = entityManager.getEntitiesWith(ClientComponent.class);
+            for (EntityRef client : clients) {
+                client.send(new TimerEvent());
+            }
+        }
+    }
+
+    private void sendGameStartMessageEventToClients() {
```

Comment:
These `send...ToClients` helper functions have the same structure and purpose, the only difference is which event they are sending. 

You could refactor this to add the event [supplier](https://docs.oracle.com/javase/8/docs/api/java/util/function/Supplier.html) as an argument here:

```java
void sendEventToClients(Supplier<Event> eventSupplier) {
        Iterable<EntityRef> clients = entityManager.getEntitiesWith(ClientComponent.class);
        for (EntityRef client : clients) {
            client.send(eventSupplier.get());
        }
}
```

On caller side this would look like this:

```java
sendEventToClients(GameStartMessageEvent::new);
// ... which is syntactic sugar for 
sendEventToClients(() -> new GameStartMessageEvent());
```